lib/rucio/rse/protocols/posix.py

```diff
@@ -210,3 +211,46 @@ def stat(self, pfn):
         """
         path = self.pfn2path(pfn)
         return {'filesize': os.stat(path)[os.path.stat.ST_SIZE], 'adler32': adler32(path)}
+
+
+class Symlink(Default):
+    """ Implementing access to RSEs using the local filesystem, creating a symlink on a get """
+
+    def get(self, pfn, dest, transfer_timeout=None):
+        """ Provides access to files stored inside connected the RSE.
+            A download/get will create a symlink on the local file system pointing to the
+            underlying file. Other operations act directly on the remote file.
+            :param pfn: Physical file name of requested file
+            :param dest: Name and path of the files when stored at the client
+            :param transfer_timeout Transfer timeout (in seconds) - dummy
+            :raises DestinationNotAccessible: if the destination storage was not accessible.
+            :raises ServiceUnavailable: if some generic error occured in the library.
+            :raises SourceNotFound: if the source file was not found on the referred storage.
+         """
+        try:
+            path = self.pfn2path(pfn)
+            os.symlink(path, dest)
+            self.logger(logging.DEBUG,
+                        'Symlink {} created for {} from {}'
+                        .format(dest, path, pfn))
+        except Exception as e:
+            print("JW,", e)
+            # catch any exceptional error
+            raise exception.ServiceUnavailable(e)
+        if not os.lstat(dest):
+            # problem in creating the symlink
+            self.logger(logging.ERROR, 'Symlink {} could not be created'.format(dest))
+            raise exception.DestinationNotAccessible()
+        if not os.path.exists(dest):
+            # could not find the file following the symlink
+            self.logger(logging.ERROR, 'Symlink {} appears to be a broken link to {}'
+                        .format(dest, path))
+            if os.lstat(dest) and os.path.islink(dest):
+                call(['rm', '-rf', dest])
```

Comment:
this can't be done with shutil.rmtree() ?

calling rm -rf seems so ... :scream:

Reply:
I was attempting consistency with (in the Default implementation): https://github.com/rucio/rucio/blob/2c9d77f203fbe6e6ec78b415ccfe457d206b0519/lib/rucio/rse/protocols/posix.py#L77 

shutil.rmtree() does not work with symlinks (afaik).  
os.unlink(dest) might be the simplest option here, (and I've put in this commit)? 

I could replace the 'rm' in the Default class implementation with an shutil.rmtree, if you wanted here?